timesketch/lib/analyzers/interface.py

```diff
@@ -401,6 +430,102 @@ def get_all_indices(self):
         return indices
 
 
+class AggregationGroup(object):
+    """Aggregation Group object with helper methods.
+
+    Attributes:
+        group (SQLAlchemy): Instance of a SQLAlchemy AggregationGroup object.
+    """
+    def __init__(self, aggregation_group):
+        """Initializes the AggregationGroup object.
+
+        Args:
+            aggregation_group: SQLAlchemy AggregationGroup object.
+        """
+        self.group = aggregation_group
+        self._how = 'layer'
+        self._parameters = ''
+
+    @property
+    def id(self):
+        """Returns the group ID."""
+        return self.group.id
+
+    @property
+    def name(self):
+        """Returns the group name."""
+        return self.group.name
+
+    def add_aggregation(self, aggregation_obj):
+        """Add an aggregation object to the group.
+
+        Args:
+            aggregation_obj (Aggregation): the Aggregation objec.
+        """
+        self.group.aggregations.append(aggregation_obj)
+        self.group.how = self._how
+        db_session.add(aggregation_obj)
+        db_session.add(self.group)
+        db_session.commit()
+
+    def commit(self):
+        """Commit changes to DB."""
+        self.group.how = self._how
+        self.group.parameters = self._parameters
+        db_session.add(self.group)
+        db_session.commit()
+
+    def set_how(self, how='layer'):
+        """Sets how charts should be joined.
+
+        Args:
```

Comment:
aha, layered as well..

I still think "how" is a little difficult to understand. Maybe "combine_as" or something similar?

Reply:
or, "layered_as" ?

Reply:
I changed how into orientation, hopefully that has "saved some troubles"